app/controllers/organizations/new.js

```diff
@@ -89,92 +135,67 @@ export default class OrganizationsNewController extends Controller {
-        organizationClassificationCode.id
-      );
+  #organizationConverter(classificationCode) {
+    // Create new model instance based on the provided classification
+    let newOrganizationModelInstance = this.#createNewModelInstance(
+      classificationCode?.id
+    );
 
-      // Copy attributes and relationships to new model instance
-      this.#copyPropertiesToModel(newOrganizationModelInstance);
-      // Note: explicitly set here to ensure form is updated
-      newOrganizationModelInstance.classification =
-        organizationClassificationCode;
-
-      // Delete the old model instance
-      // Note: this sometimes causes an InternalError concerning too much
-      // recursion to be thrown when the ember-inspector plugin is open. The
-      // error seems to be specific to the plugin and does not seem to occur
-      // otherwise.
-      this.currentOrganizationModel.deleteRecord();
-
-      // Update the organization model instance used by this controller
-      this.currentOrganizationModel = newOrganizationModelInstance;
-    }
+    // Copy attributes and relationships to new model instance
+    this.#copyPropertiesToModel(newOrganizationModelInstance);
+    // Note: explicitly set here to ensure form is updated
+    newOrganizationModelInstance.classification = classificationCode;
+
+    // Delete the old model instance
+    // Note: this sometimes causes an InternalError concerning too much
+    // recursion to be thrown when the ember-inspector plugin is open. The
+    // error seems to be specific to the plugin and does not seem to occur
+    // otherwise.
+    this.currentOrganizationModel.deleteRecord();
+
+    // Update the organization model instance used by this controller
+    this.currentOrganizationModel = newOrganizationModelInstance;
   }
 
   /**
    * Create new organization model instance of the model type that matches the
-   * provided classification code.
+   * set {@link organizationType} and/or the provided classification code.
+   * If the current user has a non-worship role the correct model type is
+   * determined based on the value of {@link organizationType}, defaulting to
+   * the most general model if no value is set.
+   * Otherwise, if the current user has a worship role, only the provided
+   * classification code identifier is used to determine the correct model
+   * type. The {@link organizationType} field is *not* used in the worship
+   * module and should never have a value.
    *
-   * @param {string} classificationCodeId - the unique identifier of
-   * the selected classification code
+   * @param {string} [classificationCodeId] - the unique identifier of the
+   *     selected classification code.
    * @returns {OrganizationModel} New model instance that matches to provided
-   * classification code
+   *     classification code.
    */
   #createNewModelInstance(classificationCodeId) {
-    // FIXME: This logic is somewhat duplicate with the classification getters
```

Comment:
**Suggestion:** In order to simplify, instead of all these if/else, can we move the `CLASSIFICATION` to model link in the `app/models/administrative-unit-classification-code.js` configuration file.

Reply:
Let's keep this for a follow up as it is a broader refactoring than just for this form.